starter/spec.js

```diff
@@ -1,9 +1,195 @@
+                usernameField.sendKeys(usernameLogin);
+            });
+            passwordField.clear().then(function(){
+                passwordField.sendKeys(passwordLogin);
+            });
+           // Press login button and verify that login is successful verifying the Url
+            // of the new page displayed
+            signInButton.click().then(function(){
+                browser.getCurrentUrl().then(function (text){
+                    expect(text).toEqual(homepageUrl);
+                });
+            })
+
+        })
+    });
+
+    describe('Logged in Homepage', function () {
+        it('should display the logged in home', function () {
+            //Check if the homepage is displayed - verifying the page Url
+            browser.getCurrentUrl().then(function (text){
+                expect(text).toEqual(homepageUrl);
+            });
+        });
+        it('should click on Content Services and go to files page', function(){
+            //go to the Files page and confirming that with the Url
+            contentServicesLink.click().then(function(){
+                browser.getCurrentUrl().then(function (text){
+                    expect(text).toEqual(filesUrl);
+                });
+            })
+        })
+    });
+
+    describe('Files page', function(){
+        it('should display the Files page', function () {
+            //Check if the Files page is displayed - verifying the page Url
+            browser.getCurrentUrl().then(function (text){
+                expect(text).toEqual(filesUrl);
+            });
+        });
+        it('should pop-up a new folder dialog box', function(){
+            //Clicks on "create new folder" icon and checks that a dialog box is displayed
+            createNewFolderIcon.click().then(function(){
+                browser.wait(EC.presenceOf(newFolderDialogBox), 5000).then(function(){
+                    expect((newFolderDialogBox).isDisplayed()).toBe(true);
+                })
+            })
+        });
+        it('should create a new folder', function () {
+            //Set the name for new folder
+            browser.wait(EC.elementToBeClickable((gitHubUsernameField), 5000)).then(function(){
+                gitHubUsernameField.click().then(function(){
+                    gitHubUsernameField.sendKeys(githubUsername);
+                });
+            });
+            //Create the folder and verify that the dialog box is not displayed anymore
+            createButton.click().then(function(){
+                browser.wait(EC.invisibilityOf(newFolderDialogBox), 5000);
+                expect((newFolderDialogBox).isPresent()).toBe(false);
+            })
+        });
+        it('should displayed my new abc folder', function(){
+            //Checks if the new folder is created
+            expect(newFolderName.isDisplayed()).toBe(true);
+        });
+
+        // it('should check for a duplicate message', function(){
```

Comment:
Can this test be uncommented and get it working. Maybe it's actually failing in the next test, because the new folder dialog is not closed?

Reply:
I've simplified the test and got it to work.